Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -192,22 +181,26 @@ public struct ConvertAction: Action, RecreatingContext {
         }
         
         self.diagnosticEngine = engine
-        self.context = try context ?? DocumentationContext(dataProvider: workspace, diagnosticEngine: engine)
         self.diagnosticLevel = filterLevel
-        self.context.externalMetadata.diagnosticLevel = self.diagnosticLevel
-        self.context.linkResolver.dependencyArchives = dependencies
         
+        var configuration = context?.configuration ?? DocumentationContext.Configuration()
+        
+        configuration.externalMetadata.diagnosticLevel = filterLevel
         // Inject current platform versions if provided
-        if let currentPlatforms {
-            self.context.externalMetadata.currentPlatforms = currentPlatforms
+        if var currentPlatforms {
+            // Add missing platforms if their fallback platform is present.
+            for (platform, fallbackPlatform) in DefaultAvailability.fallbackPlatforms where currentPlatforms[platform.displayName] == nil {
+                currentPlatforms[platform.displayName] = currentPlatforms[fallbackPlatform.displayName]
+            }
```

Comment:
This loop comes from [here](https://github.com/swiftlang/swift-docc/pull/1031/files#diff-97d66b3827022e24c102721c9f84a35bb2f5e081d80e8804cf5773dbeb8f0663L169-L175). 

In the old code it looked like setting the current platforms here did something but that value was actually written over by the `DocumentationConverter` that the `ConvertAction` creates at the end of this initializer (and passes a shared reference to the context to).